short_answer/short_answer.py

```diff
@@ -342,19 +342,25 @@ def get_submissions_list(self):
                     'max_grade': self.weight,
                     'module_type': self.category,
                     'state': '{}',
+                    'grade': 0,
                 }
             )
             state = json.loads(module.state) if module.state else {}
             state_answered_at = state.get('answered_at', '')
 
+            # For answers not defaulted to zero.
+            if module.grade is None:
+                module.grade = 0
+                module.save()
```

Comment:
I don't think we should change database state from a get method. This might cause issues in the future, in case we need to change the way this feature works.

Since this seems like a front-end logic (how we want to display grade in a certain scenario), we can do it either in the student_view method when we prepare the context or in the template, but since we work with `None`, we should do it in the context (since `None` is from Python data type set and not JS native).

Reply:
This is only to convert the `StudentModule` instances that don't have `grade=0`. So technically this isn't related to the front.end but more of unifying the default states that we have in the other blocks. And if we don't want databases changes in this function we should consider moving the `StudentModule.objects.get_or_create` above.

Reply:
Data migration is more natural solution for the problem.
Also, `get_or_create` is high level solution that will not change existing records, but this code will.

I'm just worried that this will cause future bugs in case we missed something in the bigger picture.

To truly unify default states, we should rely on an API in the Django app that handles StudentModules, instead of having each XBlock working with StudentModules specific way, especially since no back-end validation exists for most of the rows.

Since this solution seems to work, we can leave it.